scripts/autonomous_cross_deduplicator.py

```diff
@@ -52,38 +68,59 @@ def has_minhash_content(directory: Path) -> bool:
             and (directory / '1.minhashes').is_file())
 
 
-def collect_processed_batches(finished_batches_dir: Path) -> list[Path]:
+def assemble_targets(input_dir: Path, output_dir: Path,
+                     from_dir: Path, upto_dir):
     """
-    Collects the directories within the finished_batches_dir
-    which have date-like names and contain the following files:
-    1.doc_ids, 1.files, 1.minhashes
+    Returns a list of tuples. The tuples contain the parameters for the
+    deduplicate_other() method:
+    The input (minhash self) dir of the target
+    The list of dirs of older batches (minhash full)
+    The output (minhash full) dir for the target.
+
+    Ignores dirs which do not contain the required files
     """
-    collected_dirs = []
-    for directory in finished_batches_dir.iterdir():
-        if re.match('^[0-9_]+$', directory.name):
-            if has_minhash_content(directory):
-                collected_dirs.append(directory)
-        else:
-            logging.info(f'Directory name {directory.name} was not date-like')
-    collected_dirs.sort()
-    logging.info(f'The following directories contain fully processed '
-                 f'batches: {", ".join(str(d) for d in collected_dirs)}')
-    return collected_dirs
-
-
-def find_batch_to_process(self_dedup_dir: Path,
-                          finished_batch_names: list[str]) -> Path:
+    list = [dir.name for dir in sorted(input_dir.iterdir())

```

Comment:
While it is true that this is not the most elegant piece of code, it is not bad at all. It is still quite decent. 

If we have the time, sure, I can refactor it...

There are two unrelated issues here:

The original version did not have the try-excepts around it, but I felt it was a nice touch, because it might not be obvious to the user that the from and upto dirs must be ones that actually exist. I can replace it with checking if it is in the array and then raising an error, which would make the code not try-except-y (and since the arrays are 80 long, performance is not an issue on that extra iteration).

Or we can change the functionallity by using < > comparision of folder names and then not having an existing folder ceases to be a source of error.

As for the index+slice vs. dropwhile and takewhiles: 
We can use that, but it means replacing two, very short, concise lines of code which contain very basic and well known functions with a single line using a less widely known function. I am not certain if that would be an improvement when it comes to code readability.
(and code readability is the only relevant factor here, performance is not an issue, we are dealing with arrays of 80 elements).

Reply:
To make it clear: I don't mind you raising an exception. What I mind is the `try - except` stuff. The fault here lies not with you but with Python for not providing a `list.find` method. If you want to stick with the current code, I propose to have a single `try-except` block, which won't allow for such fine-grained error messages, but will result in much more succinct code (basically, one line).

As for `dropwhile` and `takewhile`: first of all, we don't replace two short lines of code, but 2 lines of code and 6 lines of `try-except` boilerplate. That looks like a good exchange to me. Also, even if the functions are arguably not that well known, they do exactly as their names imply, so it wouldn't hurt readability. That said, I am not particularly keen on them either because of the lambdas involved, so let's just go with a single `try-except` block.